py/desispec/scripts/submit_prod.py

```diff
@@ -9,109 +9,302 @@
+
+    if first_night is None:
+        if 'FIRST_NIGHT' in config:
+            first_night = int(config['FIRST_NIGHT'])
+            if verbose:
+                log.info(f"Setting first_night to FIRST_NIGHT: {first_night}")
+        else:
+            if verbose:
+                log.info("Setting first_night to earliest in a normal prod: 20201214")
+            first_night = 20201214
+
+    if all_nights is None:
+        # all_nights = get_nights_in_date_range(first_night, last_night)
+        if verbose:
+            log.info("Populating all_nights with all of the nights with valid science "
+                     + f"exposures between {first_night} and {last_night} inclusive")
+        all_nights = get_all_valid_nights(first_night, last_night)
+    return sorted(all_nights)
+
 
-def submit_production(production_yaml, dry_run=False, error_if_not_available=False):
+def submit_production(production_yaml, queue_threshold=4800, dry_run_level=False):
     """
     Interprets a production_yaml file and submits the respective nights for processing
     within the defined production.
 
     Args:
         production_yaml (str): Pathname of the yaml file that defines the production.
-        dry_run (bool, optional): Default is False. Should the jobs written to the processing table actually be submitted
-            for processing.
-        error_if_not_available (bool, optional): Default is True. Raise as error if the required exposure table doesn't exist,
-            otherwise prints an error and returns.
+        queue_threshold (int): The number of jobs for the current user in the queue
+            at which the script stops submitting new jobs.
+        dry_run_level (int, optional): Default is 0. Should the jobs written to the processing table actually be submitted
+            for processing. This is passed directly to desi_proc_night.
 
     Returns:
         None.
     """
+    log = get_logger()
+    ## Load the yaml file
     if not os.path.exists(production_yaml):
-        raise IOError(f"Prod Yaml file doesn't exist: {production_yaml} not found. Exiting.")
-    conf = yaml.safe_load(open(production_yaml, 'rb'))
-    specprod = str(conf['name']).lower()
-    specprod = verify_variable_with_environment(var=specprod, var_name='specprod', env_name='SPECPROD')
-    if 'reservation' in conf:
-        reservation = str(conf['reservation'])
+        raise IOError(f"Prod yaml file doesn't exist: {production_yaml} not found.")
+    with open(production_yaml, 'rb') as yamlfile:
+        conf = yaml.safe_load(yamlfile)
+
+    ## Make sure the specprod matches, if not set it to that in the file
+    if 'SPECPROD' not in conf:
+        raise ValueError(f"SPECPROD required in yaml file {production_yaml}")
+    specprod = str(conf['SPECPROD']).lower()
+    specprod = verify_variable_with_environment(var=specprod, var_name='specprod',
+                                                env_name='SPECPROD')
+    user = os.environ['USER']
```

Comment:
I see here that you get `user=os.environ['USER']` to later pass to `check_queue_count(user=user, ...)`, but I think `check_queue_count` itself should also have that as the default if `user=None`.

Reply:
Done